tinygrad/engine/realize.py

```diff
@@ -125,10 +125,11 @@ def __init__(self, total_sz, dest_device, src_device):
     else: name = f"{type(self).__name__[6:].lower()} {total_sz:8d}, {dest_device[:7]:>7s} <- {src_device[:7]:7s}"
     super().__init__(colored(name, "yellow"), dest_device, 0, total_sz)
   def copy(self, dest, src):
-    disk_supports_fast_copyout = src.device.startswith("DISK") and hasattr(src.allocator.device, 'io_uring') and hasattr(src.allocator.device, 'fd')
-    if src.device.startswith("DISK") and hasattr(dest.allocator, 'copy_from_disk') and disk_supports_fast_copyout and src.nbytes >= 4096:
+    is_disk = src.device.startswith("DISK")
+    disk_supports_fast_copyout = is_disk and hasattr(src.allocator.device, 'io_uring') and getattr(src.allocator.device, 'fd', None) is not None
+    if is_disk and hasattr(dest.allocator, 'copy_from_disk') and disk_supports_fast_copyout and src.nbytes >= 4096:
       dest.allocator.copy_from_disk(dest._buf, src._buf, src.nbytes)
-    elif src.device.startswith("DISK") and hasattr(dest.allocator, 'as_buffer'):
+    elif is_disk and hasattr(dest.allocator, 'as_buffer'):
```

Comment:
I factored out `is_disk = src.device.startswith("DISK")`, because otherwise the line for `disk_supports_fast_copyout` would have breached our 150 column limit.

Reply:
just use \ i can't read this diff!

minimal PRs only

Reply:
Collapsed the diff in this file to 1 line of code.